### ISSUE TITLE
[BUG] Running `ml4w-hyprland-setup` after ml4w aur update (to "ugprade) broke my touchpad functionality

### ISSUE THREAD

**Issue by armenr** [mylinuxforwork/dotfiles#268]
**Describe the bug**
After running paru and upgrading the package version for ml4w (specifically ml4w-git), I run `ml4w-hyprland-setup` --> After upgrading, two things are broken: 

1. My keyboard mapping is all off --> I go in and manually restore the keyboard settings file to the last previous version.
2. My touch pad stops functioning as it was functioning before. By this, I mean that using "touch gestures" like two-finger-swipe to switch between work spaces no longer works, etc.

Note: I'm on a Lenovo Yoga Pro laptop.  

I wonder if that's because the "post-install" script doesn't run after I run `ml4w-hyprland-setup` and reboot.

**To Reproduce**
1. working installation of ml4w with the additional setup/configs (after the first reboot of installation) with working keyboard mapping, keys, function keys, etc, + working touch pad gestures
2. Upgrade the ML4W from AUR
3. Run `ml4w-hyprland-setup` --> go through install flow
4. Reboot
5. Keys are incorrectly mapped on my laptop's keyboard (various keys are mismatched, function keys don't work)
6. Manually restore the last keyboard config file
7. Touch pad no longer recognizes multi-touch gestures like multi-finger swipe to left or right direction for switching work spaces (and I still haven't figured out how to fix that)

**Expected behavior**
Be able to correctly re-run `ml4w-hyprland-setup` without it breaking the already-working keyboard/keymappings + touch gestures on my laptop 

**Distribution (please complete the following information):**
 - CachyOS (Arch Linux)

**Comment by mylinuxforwork** [comment]
@armenr In which file have you added your custom keyboard mappings?

**Comment by mylinuxforwork** [comment]
@armenr Can you check that you have selected this layout variation in the settings app? 
https://github.com/mylinuxforwork/dotfiles/blob/main/share/dotfiles/.config/hypr/conf/layouts/laptop.conf

**Comment by mylinuxforwork** [comment]
@armenr Any updates on this?

**Comment by elliottophellia** [comment]
I also got that gesture-swipe bug after update; now moving between workspaces is a little odd without it.

**Comment by rlovell3** [comment]
Just as a passing note, I had one of my arch/ml4w machines complaining about its version of ml4w, so I ran the ml4w updater shell script.  That totally threw my keyboard into a tizzy, (possibly German?), making even the text in a terminal useless for entering commands.  So,.... I completely uninstalled ml4w and did a clean reinstall.  I'm sure there was an easier way to fix the keyboard, but the keyboard default.conf and everything else I could find had it listed as my correct"US" keyboard, so there was no easy way out. When I couldn't operate in the shell env due to the keys being off, I had to throw in the towel. This was on a desktop, thus I did not have the touchpad issue.

**Comment by rn00gr** [comment]
> @armenr Can you check that you have selected this layout variation in the settings app? https://github.com/mylinuxforwork/dotfiles/blob/main/share/dotfiles/.config/hypr/conf/layouts/laptop.conf


@mylinuxforwork  i received a notification to run "ml4w-hyprland-setup" to update to the newer version but after update i got the same problem, workspace gesture isn't working anymore. It was working fine before the update.

![screenshot_04112024_183544](https://github.com/user-attachments/assets/84288d57-0a9d-4ee9-86c6-e2e7dfa884fe)
![screenshot_04112024_183608](https://github.com/user-attachments/assets/fe198b58-30be-40f0-8255-125a268e32b5)


I do have laptop.conf file in my layouts folder. And I tried to find layout in settings app as you instructed but i can't find anything says layout in setting app.

Sir, please help.

**Comment by Forge64** [comment]
I came here to ask about this. I've been manually merging my changes back into default.conf every update, in order to get workspace swipe change working again, but it's a hassle and there doesn't seem to be anywhere in the user-facing applets to select layout. I know this used to be done in the pre/post installers when I was running the git version, but when I moved to the AUR package, that seems to have been lost.

**Comment by rn00gr** [comment]
I fixed my swap gesture issue. It wasn't anything big, I didn't notice earlier that value of swap_gesture in default.conf in layouts was set to false after the update so, i just switch it back to true and now it's working fine.

**Comment by Forge64** [comment]
> I fixed my swap gesture issue. It wasn't anything big, I didn't notice earlier that value of swap_gesture in default.conf in layouts was set to false after the update so, i just switch it back to true and now it's working fine.

That works, but only until the next update, when it gets clobbered by the defaults again. Likewise, you can change layout.conf in the ~/.config/hypr/conf/ directory to point to your customized conf, but again, it reverts on next update. All the other configs in that section have some way to specify an alternate or custom config, but layout seems to have been partially locked in, once you switch to the AUR package. The original git version of ML4W asks in the post-install if you are on a laptop and swaps the layout.conf setup. The AUR version doesn't/can't.

**Comment by mylinuxforwork** [comment]
@Forge64 Have you run restore during the update process with ml4w-hyprland-setup?

**Comment by Forge64** [comment]
@mylinuxforwork I have, with layout.conf pointing to default (modified) or machinename.conf, after update it reverts to the default with no modifications, which provides swipe=false, breaking touchpad swipe. This occurs on both git ml4w migrated to the AUR package, and new AUR-only installs, but does not affect my one remaining git install which is not using the AUR package.

**Comment by rn00gr** [comment]
yeah that's true. It always change back to false after every update.
![screenshot_12112024_220332](https://github.com/user-attachments/assets/38fb63fa-1c2b-448b-9ee6-1d0b7fc804e9)

**Comment by Forge64** [comment]
I noticed that ml4w-hyprland-setup in 2.9.7.0 now asks the "laptop" question again. Not a perfect/ideal solution, since I also use a touchpad on my desktop, but having any option helps a lot. Thanks!
